app/controllers/api/pxe_servers_controller.rb

```diff
@@ -55,6 +55,13 @@ def edit_resource(type, id, data)
       end
     end
 
+    def verify_credentials_resource(_type, _id = nil, data = {})
+      task_id = PxeServer.verify_depot_settings_queue(User.current_user.userid, data)
```

Comment:
@lfu if you're re-verifying credentials (e.g. if you edit an existing resource) we don't require the user to enter a password again but instead load it from vmdb if it doesn't exist (e.g. they only change the username or IP).

We do this by passing in `data["id"] = id if id` here and handling it on the `verify_depot_settings` side if an `"id"` is passed.

I don't know if that's something that we want to do on the UI for PxeServers but I expect it probably is and would keep parity with provider authentication verification.

Reply:
@lfu (or @Fryguy @bdunne) do you need to pass PxeServers a zone?  I don't see one in the table right now but with any "local" resource it might not be reachable from every zone.

Reply:
That's a good idea

Reply:
> @lfu if you're re-verifying credentials (e.g. if you edit an existing resource) we don't require the user to enter a password again but instead load it from vmdb if it doesn't exist (e.g. they only change the username or IP).

@agrare  `verify_depot_settings` always creates a [new instance](https://github.com/ManageIQ/manageiq/blob/de4c23643816b171243b26c5d48dc03add943873/app/models/mixins/file_depot_mixin.rb#L30) to verify the settings.

Reply:
@lfu yes the problem with that is since we don't pull passwords back over the API if the user is editing an existing instance they have to re-type their password even if they don't intend on changing it.

We can handle that in a follow-up if you want but I don't think that's the best user experience.

Reply:
@agrare Do we have a similar pattern for this with provider validation you can point to?  I think this probably is big enough that we should fix for this PR.

Reply:
Oh sorry ... didn't realize this was an outdated comment.  Is this done then?

Reply:
Mostly asking because I don't see anything that looks up the existing instance on the core side's method - https://github.com/ManageIQ/manageiq/pull/21013

Reply:
@Fryguy the API now matches the pattern for provider validation, core still doesn't lookup the instance yet not sure if you want that for this PR set or as a follow-up

Reply:
I think we need the core change then.  Otherwise, we have a "broken" API in the interim.